packages/grafana-prometheus/src/components/metrics-browser/useMetricsLabelsValues.ts

```diff
@@ -23,6 +26,9 @@ export const useMetricsLabelsValues = (timeRange: TimeRange, languageProvider: P
   const [labelValues, setLabelValues] = useState<Record<string, string[]>>({});
   const [selectedLabelValues, setSelectedLabelValues] = useState<Record<string, string[]>>({});
 
+  // Memoize the effective series limit to use the default when seriesLimit is empty
+  const effectiveLimit = useMemo(() => seriesLimit || DEFAULT_SERIES_LIMIT, [seriesLimit]);
```

Comment:
We fallback to default limit when user removed the limit at all. We can set it as `0` but that means there won't be any limit and prometheus will return everything without limit. I think we don't want that.

Reply:
If we are changing behaviour, we should also update the description in the UI, as it is not correct anymore:

``` 
Limit
Set to 'none' to remove limit and show all labels for a selected metric. Removing the limit may cause performance issues.
```

<img width="576" alt="image" src="https://github.com/user-attachments/assets/da9f7bbc-639b-4e4b-a99a-21e7c3a28c51" />